ccf_transaction_processor/Readme.md

```diff
@@ -6,37 +6,123 @@ https://creativecommons.org/licenses/by/4.0/
+before executing the above comamnd:
+
+        A. In the setup_nosgx.sh file, modify the command `sudo add-apt-repository ppa:ansible/ansible -y` by adding the `-E` option.
+           The modified command looks like `sudo -E add-apt-repository ppa:ansible/ansible -y`. It may also be necesssary to add a `sudo`
+           prefix to the `ansible-playbook ccf-dependencies-no-driver.yml` command appearing in the same file.
+
+        B. In ccf-dependencies-no-driver.yml and ccf-dependencies.yml files (present under the same folder), add the environment option for proxies
+           (see https://docs.ansible.com/ansible/latest/user_guide/playbooks_environment.html for reference)
+
+                environment:
+                    http_proxy: <specify-http-proxy-here>
+                    https_proxy: <specify-https-proxy-here>
+
+#Build CCF & PDO-TP
+
+Copy the transaction processor source files to the CCF src/apps/ folder
+
+mkdir $PDO_SRC/ccf_transaction_processor/CCF/src/apps/pdo_tp
+cp $PDO_SRC/ccf_transaction_processor/transaction_processor/*.* $PDO_SRC/ccf_transaction_processor/CCF/src/apps/pdo_tp/
+
+CCF uses a combination of cmake& ninja to build the applcaition. Execute the following steps to complete the build process:
+
+        A. Add the following lines to the CMakelists.txt found at $PDO_SRC/ccf_transaction_processor/CCF.
+           (look for other add_ccf_app in this file, add just above or below this)
+
+                add_ccf_app(
+                  pdoenc SRCS src/apps/pdo_tp/pdo_tp.cpp
+                  src/apps/pdo_tp/verify_signatures.cpp
+                )
+
+        B. Create the build folder
+
+                mkdir $PDO_SRC/ccf_transaction_processor/CCF/build
+
+        C. Set the build flags
+
+                $PDO_SRC/ccf_transaction_processor/CCF/build
+                cmake -GNinja -DCOMPILE_TARGETS=virtual -DBUILD_END_TO_END_TESTS=OFF -DBUILD_SMALLBANK=OFF
+                              -DBUILD_TESTS=OFF-DBUILD_UNIT_TESTS=OFF ..
+
+
+        D. Build using ninja
+
+                ninja
+
+           If the VM has a small memory (< 4GB), it might be required to reduce the parallelism in the build process using the `-j` flag for
+           the `ninja` command, say ninja -j2 or even ninja -j1 (the default seems to be ninja -j4).
+
+
+If build is successful, the libpdoenc.virtual.so library is created inside the build folder (ignore any other targets that get created)
+
+
+# Deploy CCF with PDO-TP as the application
+
+The following commands starts a single node CCF under the virtual enclave mode.  PDO-TP is the hosted application.
+
+cd $PDO_SRC/ccf_transaction_processor/CCF/build
+python3.7 -m venv env
+source env/bin/activate
+pip install -q -U -r ../tests/requirements.txt
+python ../tests/start_network.py --gov-script ../src/runtime_config/gov.lua  --label pdo_tp -e virtual --package libpdoenc.virtual.so --node <ip-address:6006>
```

Comment:
You want to mark up these commands as proper commands; right now they render as a (completely unparseable) single text paragraph

Reply:
sure, thank you, will fix this.